File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res = []
        for i in range(1,n+1):
            if i == 1:
                res.append(1)
            elif i == 2:
                res.append(2)
            else:
                res.append(res[-1] + res[-2])
        
        return res[-1]

File: Data Structures & Algorithms/asteroid-collision/submission-0.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Deque<Integer> stack = new ArrayDeque<>();
    for (int a : asteroids) {
        boolean alive = true;
        while (alive && a < 0 && !stack.isEmpty() && stack.peek() > 0) {
            if (stack.peek() < -a) {
                stack.pop();          // top explodes, keep checking
            } else if (stack.peek() == -a) {
                stack.pop();          // both explode
                alive = false;
            } else {
                alive = false;        // incoming explodes
            }
        }
        if (alive) stack.push(a);
    }
    int[] res = new int[stack.size()];
    for (int i = res.length - 1; i >= 0; i--) res[i] = stack.pop();
    return res;
    }
}